### PR TITLE
feat(usage): additional services price

### PR DESCRIPTION
**Pull request by radireddy**
show additional services price on monthly and hourly usage UI

Closes #MBP-286

### Requirements

* Filling out the template is required. Any pull request that does not include enough information to be reviewed in a timely manner may be closed at the maintainers' discretion.
* All new code requires tests to ensure against regressions

## Title of the Pull Requests 
Show additional services price 

### Description of the Change
When customer deploys platform like bigdata on PIC, additional money is charged to the customers. I have made UI changes to show these additional services price on monthly and hourly usage page. 

### Applicable Issues
MBP-286

### CODE REVIEW

**Comment by ovh-cds** [comment]
CDS Report deploy_uxci_2#1162.0 ✘  
* deploy_uxci
  * deploy_uxci ✘

File: client/components/cloud/project/billing/monthly-resource-list/billing-monthly-resource-list.component.html [hunk 1]
```diff
@@ -19,4 +20,19 @@
...
+            <uib-accordion-heading>
+                <button type="button" class="cloud-billing-accordion-heading" data-ng-click="$ctrl.toggleAccordion()">
+                    <span class="chevron-right">
+                        <i class="glyphicon pull-left" data-ng-class="{ 'glyphicon-chevron-down': $ctrl.toggle.accordions.additionalServices, 'glyphicon-chevron-right': !$ctrl.toggle.accordions.additionalServices }"></i>
+                    </span>
+                    <span class="cloud-billing-accordion-heading-title" data-translate="cpbc_hourly_additional_services_detail_title"></span>
+                    <span class="cloud-billing-accordion-heading-price" data-ng-bind="$ctrl.resources.totals.monthly.additionalServices.toFixed(2) + ' ' + $ctrl.resources.totals.currencySymbol"></span>
+                </button>
+            </uib-accordion-heading>
+            <additional-services services="$ctrl.resources.monthlyAdditionalServices" currency-symbol="$ctrl.resources.totals.currencySymbol"></additional-services>
```
Comment (antleblanc) [inline client/components/cloud/project/billing/monthly-resource-list/billing-monthly-resource-list.component.html]: You can add some `data-` prefixes.

File: client/components/cloud/project/billing/hourly-resource-list/billing-hourly-resource-list.component.html [hunk 2]
```diff
@@ -82,6 +87,22 @@
...
+            <uib-accordion-heading>
+                <button type="button" class="cloud-billing-accordion-heading" data-ng-click="$ctrl.toggleAccordion()">
+                    <span class="chevron-right">
+                        <i class="glyphicon pull-left" data-ng-class="{ 'glyphicon-chevron-down': $ctrl.toggle.accordions.additionalServices, 'glyphicon-chevron-right': !$ctrl.toggle.accordions.additionalServices }"></i>
+                    </span>
+                    <span class="cloud-billing-accordion-heading-title" data-translate="cpbc_hourly_additional_services_detail_title"></span>
+                    <span class="cloud-billing-accordion-heading-price" data-ng-bind="$ctrl.resources.totals.hourly.additionalServices.toFixed(2) + ' ' + $ctrl.resources.totals.currencySymbol"></span>
+                </button>
+            </uib-accordion-heading>
+            <additional-services services="$ctrl.resources.hourlyAdditionalServices" currency-symbol="$ctrl.resources.totals.currencySymbol"></additional-services>
```
Comment (antleblanc) [inline client/components/cloud/project/billing/hourly-resource-list/billing-hourly-resource-list.component.html]: Yan can add some `data-` prefixes.

File: client/components/cloud/project/billing/additional-services/additional-services.component.js [hunk 3]
```diff
@@ -0,0 +1,10 @@
+const AdditionalServicesComponent = {
```
Comment (antleblanc) [inline client/components/cloud/project/billing/additional-services/additional-services.component.js]: /cc @cbourgois which is actually migrating `client/components/cloud/*` to `client/app/cloud/components`.

File: client/components/cloud/project/billing/additional-services/additional-services.component.html [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+                <oui-column data-title=" ::'cpbc_volume_col_name' | translate " data-property="name">
+                </oui-column>
+                <oui-column data-title=" ::'cpbc_volume_col_consumption' | translate " data-property="totalPrice">
+                    <span data-ng-bind="$ctrl.getAditionalServicePrice($row)"></span>
+                </oui-column>
+            </oui-datagrid>
+        </uib-accordion-group>
+    </uib-accordion>
+</div>
+
```
Comment (antleblanc) [inline client/components/cloud/project/billing/additional-services/additional-services.component.html]: Extra empty lines.

File: client/components/cloud/project/billing/additional-services/additional-services.component.controller.js [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+        }),
+    });
+    return this.instances.load();
+  }
+
+  /**
+   * returns number of platforms (like bigdat platform) deployed on this PCI
+   */
+  getPlatformCount() {
+    return _.keys(this.services).length;
```
Comment (antleblanc) [inline client/components/cloud/project/billing/additional-services/additional-services.component.controller.js]: Is `size` make sense?

```suggestion
    return _.size(this.services);
```

File: client/components/cloud/project/billing/additional-services/additional-services.component.controller.js [hunk 6]
```diff
@@ -0,0 +1,90 @@
+class AdditionalServicesComponentCtrl {
```
Comment (marie-j) [inline client/components/cloud/project/billing/additional-services/additional-services.component.controller.js]:
```suggestion
export default class AdditionalServicesComponentCtrl {
```
?

File: client/components/cloud/project/billing/additional-services/additional-services.component.controller.js [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+  constructor($stateParams, $translate, CucControllerHelper,
```
Comment (marie-j) [inline client/components/cloud/project/billing/additional-services/additional-services.component.controller.js]:
```suggestion
/* @ngInject */
  constructor($stateParams, $translate, CucControllerHelper,
```

File: client/components/cloud/project/billing/additional-services/additional-services.component.controller.js [hunk 8]
```diff
@@ -0,0 +1,90 @@
...
+    this.initAccordians();
+    if (this.getPlatformCount() > 0) {
+      this.initInstances();
+    }
+  }
+
+  /**
+   * initiate UI accordian state, one accordian per platform
+   */
+  initAccordians() {
```
Comment (marie-j) [inline client/components/cloud/project/billing/additional-services/additional-services.component.controller.js]: Shouldn't it be `Accordions` ?

File: client/components/cloud/project/billing/additional-services/additional-services.component.controller.js [hunk 9]
```diff
@@ -0,0 +1,90 @@
...
+    const accordions = {};
+    _.forEach(_.keys(this.services), (key) => {
```
Comment (marie-j) [inline client/components/cloud/project/billing/additional-services/additional-services.component.controller.js]: Could `_.mapValues` work here 
```suggestion
  const accordions =   _.mapValues(this.services, () => false);
```

File: client/components/cloud/project/billing/additional-services/additional-services.component.html [hunk 10]
```diff
@@ -0,0 +1,32 @@
...
+<div data-ng-if="$ctrl.getPlatformCount() > 0">
+    <oui-spinner data-size="s" data-ng-if="$ctrl.instances.loading"></oui-spinner>
+    <uib-accordion data-ng-if="!$ctrl.instances.loading">
+        <uib-accordion-group data-ng-repeat="(platform, instances) in $ctrl.services"
+            data-is-open="$ctrl.toggle.accordions[platform]"
+            class="{{ $ctrl.toggle.accordions[platform] ? 'active' : ''}}">
+            <uib-accordion-heading>
+                <button type="button" class="cloud-billing-accordion-heading" data-ng-click="$ctrl.toggleAccordion()">
+                    <span class="chevron-right">
+                        <i class="glyphicon pull-left" data-ng-class="{ 'glyphicon-chevron-down': $ctrl.toggle.accordions[platform], 'glyphicon-chevron-right': !$ctrl.toggle.accordions[platform] }"></i>
```
Comment (marie-j) [inline client/components/cloud/project/billing/additional-services/additional-services.component.html]: `aria-hidden= true` or `aria-label` maybe ?

**Comment by jleveugle** [comment]
Hello @radireddy,
Your pull request is now closed due to inactivity.
Feel free to re-open it if you have any update.
Anna Bot for OVH UX <3

**Comment by radireddy** [comment]
@antleblanc can you please merge this into develop?

**Pull request merged**